[STRATCONN-1595]Update Action descriptions in Salesforce (Actions)

kwang122 opened the pull request:
<!-- Hello and thank you for contributing to Segment action-destinations! -->

<!-- Before opening your pull request, make sure you have added and ran unit
     tests and tested your change locally. Refer to our testing
     documentation for more information: https://github.com/segmentio/action-destinations/blob/main/docs/testing.md -->

<!-- If you have questions or issues please open a new issue or create a new discussion
     post in Github. -->

This PR updates action descriptions for Salesforce(Actions), this update makes it clearer about what each action does.

## Testing

No testing required as this is only a description change.

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [ ] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

marinhero reviewed (changes requested):
Don't you need to `bin/run generate:types` here too so the description on the type files is updated too?

kwang122 commented:
> Don't you need to `bin/run generate:types` here too so the description on the type files is updated too?

@marinhero The update is not necessary since it's the actions' description. `bin/run generate:types` won't update anything.

kdaswani commented on packages/destination-actions/src/destinations/salesforce/customObject/index.ts:
```diff
@@ -14,8 +14,7 @@ import Salesforce from '../sf-operations'
   title: 'Custom Object',
-  description:
-    "Represents a custom object, which you create to store information that's specific to your company or industry, or a standard object.",
+  description: 'Create, update, or upsert records in a custom object in Salesforce.',
```
Making a small change since technically this action can support other standard objects too. thanks Kevin!

```suggestion
  description: 'Create, update, or upsert records in any custom or standard object in Salesforce.',
```

kdaswani commented on packages/destination-actions/src/destinations/salesforce/lead/index.ts:
```diff
@@ -16,7 +16,7 @@ const OBJECT_NAME = 'Lead'
 const action: ActionDefinition<Settings, Payload> = {
   title: 'Lead',
-  description: 'Represents a prospect or lead.',
+  description: 'Create, update, or upsert prospects in Salesforce.',
```
```suggestion
  description: 'Create, update, or upsert leads in Salesforce.',
```

kdaswani reviewed (changes requested):
Made some super small changes - sorry for only catching/changing these now. I updated the paper doc too: https://paper.dropbox.com/doc/Salesforce-Actions-Action-Description-Renames-08102022--BnsbOFmCnEfEuYhCXXqTHeCtAg-ySvDtvlfsk0gYid5XdxpI. Thanks @kwang122 @marinhero!

kwang122 commented:
> Made some super small changes - sorry for only catching/changing these now. I updated the paper doc too: https://paper.dropbox.com/doc/Salesforce-Actions-Action-Description-Renames-08102022--BnsbOFmCnEfEuYhCXXqTHeCtAg-ySvDtvlfsk0gYid5XdxpI. Thanks @kwang122 @marinhero!

No worries, just updated!

kdaswani reviewed (approved):
thank you!
